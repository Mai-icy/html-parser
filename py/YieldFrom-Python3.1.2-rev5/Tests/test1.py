#
#   Test grammar and code generation
#

import dis

def g1():
	yield
	yield 42

def g2():
	yield from x

def g3():
	x = yield from x

def disgen(g):
	print("---------- %s ----------" % g.__name__)
	dis.dis(g)

disgen(g1)
disgen(g2)
disgen(g3)
