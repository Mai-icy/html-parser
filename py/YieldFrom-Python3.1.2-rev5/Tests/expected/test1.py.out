---------- g1 ----------
  8           0 LOAD_CONST               0 (None) 
              3 YIELD_VALUE          
              4 POP_TOP              

  9           5 LOAD_CONST               1 (42) 
              8 YIELD_VALUE          
              9 POP_TOP              
             10 LOAD_CONST               0 (None) 
             13 RETURN_VALUE         
---------- g2 ----------
 12           0 LOAD_GLOBAL              0 (x) 
              3 YIELD_FROM           
              4 POP_TOP              
              5 LOAD_CONST               0 (None) 
              8 RETURN_VALUE         
---------- g3 ----------
 15           0 LOAD_FAST                0 (x) 
              3 YIELD_FROM           
              4 STORE_FAST               0 (x) 
              7 LOAD_CONST               0 (None) 
             10 RETURN_VALUE         
